x-pack/plugin/watcher/src/test/java/org/elasticsearch/xpack/watcher/actions/email/EmailActionTests.java

```diff
@@ -605,4 +607,18 @@ private EmailAttachments randomEmailAttachments() throws IOException {
 
         return new EmailAttachments(attachments);
     }
+
+    public static class NoopEmailService extends EmailService {
```

Comment:
minor duplication here to avoid sharing classpath's between test and javaRestTests

Reply:
Could this go in the "common" qa project to be shared then?

Reply:
i would prefer not to have the unit tests depend on the qa project (which has the parent classes for the java and yaml rest test runners). creating a dependency the other way requires a new test artifact or test fixture (in this case I prefer the minor duplication)

Reply:
No worries. It's a tiny implementation so it's probably not a big deal for a bit of duplication here.